src/DynamicData/Cache/Internal/SortAndPage.cs

```diff
@@ -0,0 +1,195 @@
+    private readonly SortAndPageOptions _options;
+
+    public SortAndPage(IObservable<IChangeSet<TObject, TKey>> source,
+        IComparer<TObject> comparer,
+        IObservable<IPageRequest> virtualRequests,
+        SortAndPageOptions options)
+        : this(source, Observable.Return(comparer), virtualRequests, options)
+    {
+    }
+
+    public SortAndPage(IObservable<IChangeSet<TObject, TKey>> source,
+        IObservable<IComparer<TObject>> comparerChanged,
+        IObservable<IPageRequest> virtualRequests,
+        SortAndPageOptions options)
+    {
+        _options = options;
+        _source = source ?? throw new ArgumentNullException(nameof(source));
+        _comparerChanged = comparerChanged;
+        _pageRequests = virtualRequests ?? throw new ArgumentNullException(nameof(virtualRequests));
+    }
+
+    private static readonly ChangeSet<TObject, TKey, PageContext<TObject>> Empty = new(0, PageContext<TObject>.Empty);
+
+    public IObservable<IChangeSet<TObject, TKey, PageContext<TObject>>> Run() =>
+        Observable.Create<IChangeSet<TObject, TKey, PageContext<TObject>>>(
+            observer =>
+            {
+                var locker = new object();
+
+                var sortOptions = new SortAndBindOptions
+                {
+                    UseBinarySearch = _options.UseBinarySearch,
+                    ResetThreshold = _options.ResetThreshold
+                };
+
+                var pageRequest = PageRequest.Default;
+
+                // a sorted list of key value pairs
+                var sortedList = new List<KeyValuePair<TKey, TObject>>(_options.InitialCapacity);
+                var pagedItems = new List<KeyValuePair<TKey, TObject>>(pageRequest.Size);
+
+                IComparer<TObject>? comparer = null;
+                KeyValueComparer<TObject, TKey>? keyValueComparer = null;
+                SortedKeyValueApplicator<TObject, TKey>? applicator = null;
+
+                // used to maintain a sorted list of key value pairs
+                var comparerChanged = _comparerChanged.Synchronize(locker)
+                    .Select(c =>
+                    {
+                        comparer = c;
+                        keyValueComparer = new KeyValueComparer<TObject, TKey>(c);
+
+                        if (applicator is null)
+                        {
+                            applicator = new SortedKeyValueApplicator<TObject, TKey>(sortedList, keyValueComparer, sortOptions);
+                        }
+                        else
+                        {
+                            applicator.ChangeComparer(keyValueComparer);
+                        }
+                        return ApplyPagedChanges();
+                    });
+
+                var paramsChanged = _pageRequests.Synchronize(locker)
+                    .DistinctUntilChanged()
+                    // exclude dodgy params
+                    .Where(parameters => parameters is { Page: > 0, Size: > 0 })
```

Comment:
Doesn't this filter out some valid requests? E.G. `{ Page: 0, Size: 10 }`? Or is `Page` here 1-indexed?

Reply:
It just ignores bad input. The other alternative would be just to throw an exception

Reply:
Dang, I just re-read your comments. Page zero is not valid. Paging always starts at 1.